contracts/oracles/fBEETSPriceOracle.sol

```diff
@@ -2,48 +2,70 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import "./interfaces/AggregatorV3Interface.sol";
 
 import "./SpotPriceOracle.sol";
 import "./interfaces/IBalV2PriceOracle.sol";
 
 import "hardhat/console.sol";
 
+interface IFBeets {
+    function vestingToken() external view returns (address);
+}
+
 contract fBEETSPriceOracle is SpotPriceOracle {
     using SafeMath for uint256;
+    using Address for address;
 
     address vestingToken;
     address ftmUsdOracle;
     address fBeetsBar;
 
-    constructor(
-        address _vestingToken,
-        address _ftmUsdOracle,
-        address _fBeetsBar
-    ) SpotPriceOracle() {
-        vestingToken = _vestingToken;
+    constructor(address _fBeetsBar) SpotPriceOracle() {
+        require(_fBeetsBar.isContract(), "fBeetsBar is not a contract");
         ftmUsdOracle = _ftmUsdOracle;
         fBeetsBar = _fBeetsBar;
+        vestingToken = IFBeets(_fBeetsBar).vestingToken();
     }
 
-    function getPrice() public view override returns (int256) {
-        IBalV2PriceOracle.OracleAverageQuery[]
-            memory query = new IBalV2PriceOracle.OracleAverageQuery[](1);
-        query[0] = IBalV2PriceOracle.OracleAverageQuery(
-            IBalV2PriceOracle.TWAP_VALUE.BPT_PRICE,
-            120,
-            10
-        );
+    function updateSafePrice() public payable returns (int256) {
+       return getSafePriceInternal();
+    }
+
+    function getSafePrice() public view override returns (int256) {
+       return getSafePriceInternal();
+    }
+
+    function getSafePriceInternal() internal view returns (int256) {
+      IBalV2PriceOracle.OracleAverageQuery[]
+          memory query = new IBalV2PriceOracle.OracleAverageQuery[](1);
+      query[0] = IBalV2PriceOracle.OracleAverageQuery(
+          IBalV2PriceOracle.TWAP_VALUE.BPT_PRICE,
+          120,
+          10
+      );
+      uint256[] memory prices = IBalV2PriceOracle(vestingToken).getTimeWeightedAverage(
+          query
+      );
+      uint256 bptFtm = prices[0];
+      uint256 fBeetsTotalSupply = IERC20(fBeetsBar).totalSupply();
+      uint256 vestingTokenLocked = IERC20(vestingToken).balanceOf(fBeetsBar);
+      uint256 bptFBeetsRatio = vestingTokenLocked.mul(1e18).div(fBeetsTotalSupply);
+      uint256 price = bptFtm.mul(bptFBeetsRatio);//.div(1e36);
+      return int256(price);
+    }
+
+    // fast but vulnerable to flashloan attacks
```

Comment:
There is no concept of fast in smart contracts. Everything is done inside a tx that is confirmed depending on the gas the users pays. More operations generally means higher gas costs. Moreover as this function uses the Bal oracle, it is not vulnerable to flashloans